### ISSUE TITLE
[TODO] updated to nginx 1.15.10

### ISSUE THREAD

**Issue by netroby** [alibaba/tengine#1222]
```
Changes with nginx 1.15.10                                       26 Mar 2019

    *) Change: when using a hostname in the "listen" directive nginx now
       creates listening sockets for all addresses the hostname resolves to
       (previously, only the first address was used).

    *) Feature: port ranges in the "listen" directive.

    *) Feature: loading of SSL certificates and secret keys from variables.

    *) Workaround: the $ssl_server_name variable might be empty when using
       OpenSSL 1.1.1.

    *) Bugfix: nginx/Windows could not be built with Visual Studio 2015 or
       newer; the bug had appeared in 1.15.9.

```
https://github.com/nginx/nginx/releases/tag/release-1.15.10

How about merge upstream 1.15.10 code base into tengine?

**Comment by chobits** [comment]
Yes, this is still planned.

**Comment by chobits** [comment]
I am closing this issue, @MengqiWu has merged nginx-1.16.0 in this https://github.com/alibaba/tengine/pull/1293.

We'll release tengine 2.3.1 later. For more details see https://github.com/alibaba/tengine/issues/1292.
